src/exo/cursors.py

```diff
@@ -0,0 +1,599 @@
+        elif isinstance(_node, list):
+            raise ValueError('must index into block attribute')
+        cur = Node(self._proc, self._path + [(attr, i)])
+        # noinspection PyPropertyAccess
+        # cached_property is settable, bug in static analysis
+        cur._node = weakref.ref(_node)
+        return cur
+
+    def children(self) -> Iterable[Node]:
+        n = self.node()
+        # Top-level proc
+        if isinstance(n, LoopIR.proc):
+            yield from self._children_from_attrs(n, 'body')
+        # Statements
+        elif isinstance(n, (LoopIR.Assign, LoopIR.Reduce)):
+            yield from self._children_from_attrs(n, 'idx', 'rhs')
+        elif isinstance(n, (LoopIR.WriteConfig, LoopIR.WindowStmt)):
+            yield from self._children_from_attrs(n, 'rhs')
+        elif isinstance(n, (LoopIR.Pass, LoopIR.Alloc, LoopIR.Free)):
+            yield from []
+        elif isinstance(n, LoopIR.If):
+            yield from self._children_from_attrs(n, 'cond', 'body', 'orelse')
+        elif isinstance(n, (LoopIR.ForAll, LoopIR.Seq)):
+            yield from self._children_from_attrs(n, 'hi', 'body')
+        elif isinstance(n, LoopIR.Call):
+            yield from self._children_from_attrs(n, 'args')
+        # Expressions
+        elif isinstance(n, LoopIR.Read):
+            yield from self._children_from_attrs(n, 'idx')
+        elif isinstance(n, (LoopIR.Const, LoopIR.WindowExpr, LoopIR.StrideExpr,
+                            LoopIR.ReadConfig)):
+            yield from []
+        elif isinstance(n, LoopIR.USub):
+            yield from self._children_from_attrs(n, 'arg')
+        elif isinstance(n, LoopIR.BinOp):
+            yield from self._children_from_attrs(n, 'lhs', 'rhs')
+        elif isinstance(n, LoopIR.BuiltIn):
+            yield from self._children_from_attrs(n, 'args')
+
+    def _children_from_attrs(self, n, *args) -> Iterable[Node]:
+        for attr in args:
+            children = getattr(n, attr)
+            if isinstance(children, list):
+                for i in range(len(children)):
+                    yield self.child(attr, i)
+            else:
+                yield self.child(attr, None)
+
+    # ------------------------------------------------------------------------ #
+    # Navigation (block selectors)
+    # ------------------------------------------------------------------------ #
+
+    def body(self) -> Selection:
+        return self._select_attr('body')
+
+    def orelse(self) -> Selection:
+        return self._select_attr('orelse')
+
+    def _select_attr(self, attr: str):
+        stmts = getattr(self.node(), attr)
+        assert isinstance(stmts, list)
+        return Selection(self._proc, self._path + [(attr, range(len(stmts)))])
+
+    # ------------------------------------------------------------------------ #
+    # Conversions
+    # ------------------------------------------------------------------------ #
+
+    def select(self) -> Selection:
```

Comment:
Avoid "selection" name

Reply:
Maybe "selections" can be renamed as "block" or "statements"?

Reply:
I'll go with Block

Reply:
Done.

Reply:
Done.